frontend/src/layout/navigation-3000/sidepanel/panels/sidePanelDocsLogic.ts

```diff
@@ -14,32 +29,58 @@ export const sidePanelDocsLogic = kea<sidePanelDocsLogicType>([
 
     actions({
         openDocsPage: (urlOrPath: string) => ({ urlOrPath }),
+        updatePath: (path: string) => ({ path }),
+        setInitialPath: (path: string) => ({ path }),
+        unmountIframe: true,
+        handleExternalUrl: (urlOrPath: string) => ({ urlOrPath }),
     }),
 
     reducers(() => ({
-        path: [
+        currentPath: [
+            null as string | null,
+            {
+                updatePath: (_, { path }) => sanitizePath(path),
+            },
+        ],
+        initialPath: [
             '/docs' as string,
+            { persist: true },
             {
-                openDocsPage: (_, { urlOrPath }) => {
-                    let path = urlOrPath
-                    try {
-                        const url = new URL(urlOrPath)
-                        if (url.origin === POSTHOG_COM_DOMAIN) {
-                            path = url.pathname + url.search
-                        }
-                    } catch (e) {
-                        // not a valid URL, continue
-                    }
-
-                    return path[0] === '/' ? path : `/${path}`
-                },
+                setInitialPath: (_, { path }) => sanitizePath(path),
             },
         ],
     })),
 
-    listeners(({ actions }) => ({
-        openDocsPage: () => {
+    selectors({
+        iframeSrc: [
+            (s) => [s.initialPath],
+            (initialPath) => {
+                return `${POSTHOG_WEBSITE_ORIGIN}${initialPath ?? ''}`
+            },
+        ],
+        currentUrl: [
+            (s) => [s.currentPath],
+            (currentPath) => {
+                return `${POSTHOG_WEBSITE_ORIGIN}${currentPath ?? ''}`
+            },
+        ],
+    }),
+
+    listeners(({ actions, values }) => ({
+        openDocsPage: ({ urlOrPath }) => {
+            actions.setInitialPath(getPathFromUrl(urlOrPath))
             actions.openSidePanel(SidePanelTab.Docs)
         },
+
+        unmountIframe: () => {
+            // Update the initialPath so that next time we load it is the same as last time
+            actions.setInitialPath(values.currentPath ?? '/docs')
+
+            // TODO: Do we need to call this before the window unloads?
```

Comment:
TODO left.
Yes, this would need to be in a `beforeunload` handler, even though these are never 100% reliable (there are also some shenanigans with back/forward cache invalidation, but we probably don't need to care about that)

Reply:
Yeah I guess its better than nothing as this is not the most important feature